src/entity/Company.java

```diff
@@ -0,0 +1,30 @@
+package entity;
+
+import utils.CarPoolGenerator;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Company {
+    private List<Car> cars;
+    private static Company company = null;
+    private static AtomicBoolean instanceCreated = new AtomicBoolean(false);
+
+    private Company() {
+        cars = CarPoolGenerator.getCars();
```

Comment:
You shouldn't have any logic inside constructors. So I would suggest to move it in getCompany() method and pass value as a constructor argument.